src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,173 @@
+        !containerRef.current.contains(event.target as Node) &&
+        popoverContent &&
+        !popoverContent.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+        if (!sanitizedOptions.some((option) => option === inputValue.toLowerCase())) {
+          setInputValue('');
+        }
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [inputValue, sanitizedOptions, baseClassName, setInputValue]);
+
+  return (
+    <div
+      ref={(node) => {
+        containerRef.current = node;
+        if (typeof ref === 'function') ref(node);
+        else if (ref) (ref as React.MutableRefObject<HTMLDivElement | null>).current = node;
+      }}
+      className={classnames(baseClassName, className)}
+      id={id}
+      {...commonProps}
+      {...props}
+    >
+      <label htmlFor={`${id}-input`} className={`${baseClassName}__label`} data-testid={`${id}-label`}>
+        {label}
+      </label>
+      <Command
+        loop
+        shouldFilter={false}
+        onKeyDown={(e) => {
+          setTimeout(() => {
+            if (e.key === 'Escape') {
+              setIsOpen(false);
+            }
+          }, 0);
+        }}
+      >
+        <Popover.Root open={true}>
+          <Popover.Trigger asChild>
+            <CommandInput
+              placeholder={placeholder}
+              value={inputValue}
+              onValueChange={(value) => {
+                setInputValue(value);
+                setIsOpen(true);
+              }}
+              onFocus={() => {
+                setIsOpen(true);
+                setInputValue(inputValue);
+              }}
+              className={`${baseClassName}__input`}
+              tabIndex={0}
+              aria-label={`${id}-input`}
+              data-testid={`${id}-input`}
+            />
+          </Popover.Trigger>
+          {inputValue.length > 0 && (
+            <button
+              className={`${baseClassName}__close-button`}
+              data-testid={`${id}-item-close-button`}
+              onClick={() => setInputValue('')}
+              aria-label={`${id}-clear`}
+            >
+              <Icon color="$primary-black" icon="Close" height={18} width={18} className={`${baseClassName}__icon`} />
+            </button>
+          )}
+          <button
+            aria-label={`${id}-dropdown`}
+            className={`${baseClassName}__dropdown-button`}
+            onClick={() => setIsOpen((prev) => !prev)}
+            data-testid={`${id}-dropdown`}
+          >
+            <Icon color="$pure-black" height={18} icon="ChevronDown" width={18} className={`${baseClassName}__icon`} />
+          </button>
+          <Popover.Portal>
+            <Popover.Content className={`${baseClassName}__content`} aria-label={`${id}-content`}>
+              <CommandList className={`${baseClassName}__list`} hidden={!isOpen}>
```

Comment:
I'm not sure this is what we want, this keeps the command list in the dom even when it's not expanded.  can't we just have the PopoverRoot open change?

Reply:
```suggestion
      {isOpen ? (
                <CommandList className={`${baseClassName}__list`}>
```